### PR TITLE
Improve order flow storing main

### PR DESCRIPTION
**Pull request by islean**
### Added

-

### Changed

- FastqValidationService used in submission
- LIMS works with Order objects instead of OrderIn objects

### Fixed

-

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/meta/orders/ticket_handler.py [hunk 1]
```diff
@@ -212,3 +223,41 @@ def connect_to_ticket(
             )
 
             LOG.info(f"Connected order to ticket {ticket_number} in Freshdesk")
+
+    def create_case_xml_sample_list(self, order, message: str) -> str:
+        for case in order.cases:
+            if not case.is_new:
```
Comment (diitaz93) [inline cg/meta/orders/ticket_handler.py:229]: refactor this in smaller functions

File: cg/services/order_validation_service/workflows/fastq/validation_service.py [hunk 2]
```diff
@@ -20,14 +21,12 @@ def __init__(self, store: Store):
...
+        if parsed_order:
+            errors: ValidationErrors = self._perform_rule_validation(order=parsed_order)
         return create_order_validation_response(raw_order=raw_order, errors=errors)
 
-    def _get_errors(self, raw_order: dict) -> ValidationErrors:
-        order, field_errors = ModelValidator.validate(order=raw_order, model=FastqOrder)
-
-        if not order:
-            return field_errors
+    def _perform_rule_validation(self, order: FastqOrder) -> ValidationErrors:
```
Comment (diitaz93) [inline cg/services/order_validation_service/workflows/fastq/validation_service.py:29]:
```suggestion
    def _get_rule_validation_errors(self, order: FastqOrder) -> ValidationErrors:
```

File: cg/services/order_validation_service/workflows/fastq/validation_service.py [hunk 3]
```diff
@@ -20,14 +21,12 @@ def __init__(self, store: Store):
         self.store = store
 
     def validate(self, raw_order: dict) -> dict:
```
Comment (diitaz93) [inline cg/services/order_validation_service/workflows/fastq/validation_service.py:23]: ??
```suggestion
    def get_validation_response(self, raw_order: dict) -> dict:
```

File: cg/services/order_validation_service/workflows/mutant/constants.py [hunk 4]
```diff
@@ -4,7 +4,7 @@
 
 
 class MutantDeliveryType(Enum):
-    ANALYSIS_FILES = DataDelivery.FASTQ_ANALYSIS
+    ANALYSIS_FILES = DataDelivery.FASTQ
```
Comment (diitaz93) [inline cg/services/order_validation_service/workflows/mutant/constants.py:7]: This has to be changed as @ChrOertlin fixed this

File: cg/services/orders/store_order_services/store_fastq_order_service.py [hunk 5]
```diff
@@ -22,51 +21,26 @@ def __init__(self, status_db: Store, lims_service: OrderLimsService):
...
-                    "subject_id": sample.subject_id,
-                    "tumour": sample.tumour,
-                    "volume": sample.volume,
-                }
-                for sample in order.samples
-            ],
-        }
-        return status_data
+        self._fill_in_sample_ids(samples=order.samples, lims_map=lims_map)
+        new_samples = self.store_items_in_status(order=order)
```
Comment (diitaz93) [inline cg/services/orders/store_order_services/store_fastq_order_service.py:36]:
```suggestion
        new_samples: list[Sample] = self.store_items_in_status(order=order)
```

File: cg/services/orders/store_order_services/store_case_order.py [hunk 6]
```diff
@@ -52,9 +45,7 @@ def _process_case_samples(self, order: OrderIn) -> dict:
         # submit new samples to lims
         new_samples = [sample for sample in order.samples if sample.internal_id is None]
         if new_samples:
-            project_data, lims_map = self.lims.process_lims(
-                lims_order=order, new_samples=new_samples
-            )
+            project_data, lims_map = self.lims.process_lims(order=order, new_samples=new_samples)
```
Comment (diitaz93) [inline cg/services/orders/store_order_services/store_case_order.py:48]: fix signature with correct arguments

File: cg/services/order_validation_service/workflows/mutant/constants.py (hunk 4 above)
Comment (islean) [reply cg/services/order_validation_service/workflows/mutant/constants.py:7]: Has that been merged? As far as I know, we are still keeping them as fastq. Could we create an issue and revisit this when it has gone live?

Comment (islean) [reply cg/services/order_validation_service/workflows/mutant/constants.py:7]: It is listed in https://github.com/Clinical-Genomics/improve-order-flow/issues/99

File: cg/services/orders/store_order_services/store_metagenome_order.py [hunk 7]
```diff
@@ -29,7 +22,7 @@ def __init__(self, status_db: Store, lims_service: OrderLimsService):
 
     def store_order(self, order: OrderIn) -> dict:
         """Submit a batch of metagenome samples."""
-        project_data, lims_map = self.lims.process_lims(lims_order=order, new_samples=order.samples)
+        project_data, lims_map = self.lims.process_lims(order=order, new_samples=order.samples)
```
Comment (diitaz93) [inline cg/services/orders/store_order_services/store_metagenome_order.py:25]: fix signature with correct arguments

File: cg/services/orders/store_order_services/store_microbial_order.py [hunk 8]
```diff
@@ -36,12 +28,10 @@ def __init__(self, status_db: Store, lims_service: OrderLimsService):
     def store_order(self, order: OrderIn) -> dict:
         self._fill_in_sample_verified_organism(order.samples)
         # submit samples to LIMS
-        project_data, lims_map = self.lims.process_lims(lims_order=order, new_samples=order.samples)
+        project_data, lims_map = self.lims.process_lims(order=order, new_samples=order.samples)
```
Comment (diitaz93) [inline cg/services/orders/store_order_services/store_microbial_order.py:31]: fix signature with correct arguments

File: cg/services/orders/store_order_services/store_case_order.py (hunk 6 above)
Comment (islean) [reply cg/services/orders/store_order_services/store_case_order.py:48]: Is that not for the work item that reworks the case order though?

Comment (diitaz93) [reply cg/services/orders/store_order_services/store_case_order.py:48]: yes, I will resolve

File: cg/meta/orders/ticket_handler.py (hunk 1 above)
Comment (diitaz93) [reply cg/meta/orders/ticket_handler.py:180]: maybe for the main dev branch

File: cg/services/order_validation_service/workflows/fastq/validation_service.py (hunk 2 above)
Comment (diitaz93) [reply cg/services/order_validation_service/workflows/fastq/validation_service.py]: changed

**Review by diitaz93 — APPROVED** [review]
Looks complete enough to go to the main dev branch 💪

File: cg/services/order_validation_service/workflows/fastq/validation_service.py [hunk 9]
```diff
@@ -44,3 +43,11 @@ def _get_errors(self, raw_order: dict) -> ValidationErrors:
             sample_errors=sample_errors,
             order_errors=order_errors,
         )
+
+    def parse_and_validate(self, raw_order: dict) -> FastqOrder:
+        parsed_order, errors = ModelValidator.validate(order=raw_order, model=FastqOrder)
+        if parsed_order:
+            errors: ValidationErrors = self._perform_rule_validation(order=parsed_order)
+        if not errors.is_empty:
+            raise OrderSubmissionError(message="Order contained errors")
```
[the remark this answers is not in the document]
Comment (islean) [reply cg/services/order_validation_service/workflows/fastq/validation_service.py]: Changed to different error which will be caught

**Pull request merged**